reactivesocket-core/src/main/java/io/reactivesocket/internal/Responder.java

```diff
@@ -483,16 +485,16 @@ private void cleanup() {
                 }
 
             };
-            synchronized(Responder.this) {
+            synchronized(this) {
```

Comment:
Actually not, `new Subscriber` ends at line 464 and `Publisher` at 487.
This `synchronized` block belongs to `handleRequestResponse` function.

This and other changes were suggested by Intellij for me!